assets/js/background.js

```diff
@@ -178,6 +185,103 @@ function handleOnUpdated(tabId, changeInfo, tab) {
 }
 
 /**
+ * Message listenener for context requests
+ */
+var findContext = function findContext(request, sender, sendResponse) {
+  chrome.tabs.query({
+    active: true,
+    currentWindow: true
+  }, function(tabs) {
+    var tab = tabs[0];
+    var site = getSiteFromTab(config.supportedSites, tab);
+    // Unsupported website, skip.
+    if(!site) {
+      console.warn('Unsupported');
+      return;
+    }
+    // Retrieve context from tab
+    detectContextWithRetry(tab, site, 2, 1000, function(err, context) {
+      if(err) {
+        console.warn(err);
+        return;
+      }
+      sendResponse({
+        site: site,
+        context: blacklist.filterQuery(context)
+      });
+      return;
+    });
+  });
+  return true;
+};
+
+
+/**
+ * Message listenener for search requests
+ */
+var getResults = function getResults(request, sender, sendResponse) {
+  var context = request.context;
+  if(!context.length) {
+    return;
+  }
+
+  var query = generateQuery(context);
+  var payload = {
+    query: query,
+    check: request.check
+  };
+
+  async.waterfall([
+    function loadSettings(cb) {
+      if(config.loaded) {
```

Comment:
I think the config check should be done in getDocuments, not here, since config is not used here implicitly